components/Login/index.js

```diff
@@ -0,0 +1,115 @@
+import { signIn, signOut, useSession } from "next-auth/react";
+import styled from "styled-components";
+import Image from "next/image";
+import { useState } from "react";
+
+export default function Login({ isHomepage }) {
+  const { data: session } = useSession();
+  const [isHovered, setIsHovered] = useState(false);
+
+  function handleSignIn() {
+    signIn("github", { callbackUrl: "/exercises" });
+  }
+
+  const ButtonComponent = isHomepage ? HomepageButton : AnimatedButton;
+
+  return (
+    <ButtonComponent
+      onClick={session ? () => signOut() : handleSignIn}
+      onMouseEnter={() => setIsHovered(true)}
+      onMouseLeave={() => setIsHovered(false)}
+      onTouchStart={() => setIsHovered(!isHovered)}
+      $isHovered={isHovered}
+    >
+      {session ? (
+        isHomepage ? (
+          "Log Out"
+        ) : (
+          <>
+            <span className="text">Log Out</span>
+            <Image
+              alt="Log Out"
+              width={35}
+              height={35}
+              src={"/icons/account-white.svg"}
+            />
+          </>
+        )
+      ) : isHomepage ? (
+        "Log In with Github"
+      ) : (
+        <>
+          <span className="text">Log In</span>
+          <Image
+            alt="Log In"
+            width={35}
+            height={35}
+            src={"/icons/account-white.svg"}
+          />
+        </>
+      )}
+    </ButtonComponent>
```

Comment:
4. Avoid Unnecessary Fragments: Removed unnecessary fragment wrapping around within the button component


The fragments <> </>are unnecessary because there is only one root element (ButtonComponent) being returned in each case. In this scenario, the fragment doesn't serve a purpose since a single element can be returned directly without wrapping it.
By removing the <> </> wrappers, the code becomes cleaner and more concise without affecting functionality.